### PR TITLE
GPO listener client connection and device reassignment

### PR DESCRIPTION
**Pull request by syoffe**
Corrected a few errors preventing the GPO listener client connecting to the server and handling device reassignment properly. Made similar changes to `src/index.ts` for relay listener, too (untested).

### CODE REVIEW

**Comment by syoffe** [comment]
Hello @MatteoGheza,

The automatic build failure for `TTGO T7 v14 Mini32 Listener` seems to be unrelated to the code changes, with the error being installing websockets:
```Running command: /home/runner/bin/arduino-cli lib install WebSockets 
[58](https://github.com/josephdadams/TallyArbiter/actions/runs/3604854466/jobs/6083972336#step:4:62)
   Downloading WebSockets@2.3.6...
[59](https://github.com/josephdadams/TallyArbiter/actions/runs/3604854466/jobs/6083972336#step:4:63)
  
[60](https://github.com/josephdadams/TallyArbiter/actions/runs/3604854466/jobs/6083972336#step:4:64)
  WebSockets@2.3.6 0 B / 981 B    0.00%
[61](https://github.com/josephdadams/TallyArbiter/actions/runs/3604854466/jobs/6083972336#step:4:65)
  WebSockets@2.3.6 Server responded with: 503 Service Unavailable
[62](https://github.com/josephdadams/TallyArbiter/actions/runs/3604854466/jobs/6083972336#step:4:66)
  Error installing WebSockets: Can't download library: Server responded with: 503 Service Unavailable
```

**Comment by MatteoGheza** [comment]
Thanks for the contribution, and sorry for the delay...
I'll look into this PR later this day, as soon as possibile.
The CI automatic build pipeline needs to be updated, thanks for reporting this to me, even if I think this is a platformIO server-related issue.

After taking a quick look at the code, I'll add some reviews.

File: listener_clients/gpo-listener/gpo-listener.py [hunk 1]
```diff
@@ -344,7 +346,7 @@ def main():
             while True:
                 time.sleep(0.1)
         else:
-            server_connect(str(config_object["server_config"]["ip"]), str(config_object["server_config"]["port"]))
+            server_connect("http://" + str(config_object["server_config"]["ip"]) + ":" + str(config_object["server_config"]["port"]))
```
Comment (MatteoGheza) [inline listener_clients/gpo-listener/gpo-listener.py:349]: Good catch! I forgot to test this condition 🤦🏻

File: listener_clients/gpo-listener/gpo-listener.py [hunk 2]
```diff
@@ -262,7 +264,7 @@ def on_reassign(oldDeviceId, newDeviceId, gpoGroupId):
             gpo_group["deviceId"] = newDeviceId
 
     saveConfig()
-
+    sio.emit('listener_reassign_gpo', (gpoGroupId, oldDeviceId, newDeviceId));
```
Comment (MatteoGheza) [inline listener_clients/gpo-listener/gpo-listener.py:267]: We forgot to document this in the code.
We are trying to deprecate the old `listener_reassign_gpo` events, migrating to a single event for all the different types of listeners. I'll update this and add some warnings in the codebase.

Comment (syoffe) [reply listener_clients/gpo-listener/gpo-listener.py:267]: Hi @MatteoGheza, ah no problem, that makes sense. Let me know if I can help test in any way. Part of the problem with the reassign was using the variable `listener_clients[I].gpoGroupId` in `index.ts`, which should have been `listener_clients[i].internalId`. I guess this is part of the migration to a single interface for all devices.

**Pull request merged**